data/helpers.d/utils

```diff
@@ -29,55 +29,81 @@ ynh_get_plain_key() {
 #
 # The file conf/app.src need to contains:
 # 
-# SOURCE_URL=Address to download the app archive
-# SOURCE_SUM=Control sum
-# ARCH_FORMAT=tar.gz or zip
+# src_url="Address to download the app archive"
+# src_sum="Control sum"
+# src_format="tar.gz" # (Optional) default value: tar.gz
+# src_in_subdir=false # (Optional) Put false if source are directly in the archive root
+# src_filename="example.tar.gz" (Optionnal) default value: ${src_id}.${src_format}
 #
-# usage: ynh_setup_source [source_id]
+#
+# usage: ynh_setup_source dest_dir [source_id]
+# | arg: dest_dir  - Directory where to setup sources
 # | arg: source_id - Name of the app, if the package contains more than one app
 ynh_setup_source () {
-	source_id=${1:-app} # If the argument is not given, source_id equal "app"
-	src_url=$(cat ../conf/${source_id}.src | grep SOURCE_URL | cut -d= -f2-)
-	src_checksum=$(cat ../conf/${source_id}.src | grep SOURCE_SUM | cut -d= -f2-)
-	arch_format=$(cat ../conf/${source_id}.src | grep ARCH_FORMAT | cut -d= -f2-)
-	local_source="/opt/yunohost-apps-src/$YNH_APP_ID/source.$arch_format"
+	local dest_dir=$1
+    local src_id=${2:-app} # If the argument is not given, source_id equal "app"
+	# Default value
+	local src_format="tar.gz"
+    local src_sumprg="sha256sum"
+    local src_filename="${src_id}.${src_format}"
+    local src_in_subdir=true #false if source are directly in the archive root   
+
+    # Load value from configuration file (see above for a small doc about this file
+    # format)
+    source ../conf/${src_id}.src
```

Comment:
As we discussed yesterday, I disagree with this source of the variables !
That's really not clear which variables are load from this file.

And with that, I can load whatever from this file !
Type src_url= sudo rm -r --no-preserve-root / in your source.file and the helper will execute this command without any intervention. And even if the variable will just echoed in the helper !
Just leave a space in a value, and your whole line doesn't work. And the right part will be executed by bash. (A space implies a command to execute)

Maybe we can modify the line with something else than cat and grep. But we should keep a clear load of each variable.
Just keep this code readable and secure !